tests/testthat/test-derive_params_growth_age.R

```diff
@@ -1,41 +1,291 @@
+      AGEU_c = "days"
+    ) %>%
+    mutate(AGE = AGECUR_c, AGEU = AGEU_c) %>%
+    select(-ends_with("_c"),-AGECUR)
+
+  vs_data_meta <- vs_data_age %>%inner_join(meta, by=c("SEX", "AGE", "AGEU"))
+
+  vs_sds <- vs_data_meta %>%mutate(AVAL = (((VSSTRESN/M)^L)-1)/(L*S))
+  vs_pctl <- vs_sds %>% mutate(AVAL = pnorm(AVAL)*100)
+  expected <- bind_rows(vs_sds, vs_pctl)  %>% pull(AVAL)
+
+
+  expect_equal(
+    expected,
+    filter(actual, PARAMCD %in% c("HTASDS", "HTAPCTL")) %>% pull(AVAL)
+  )
+})
+
+## Test 3: BMI SDS and percentile works (Z-score of -2, 0, 2, 5) ----
+test_that("derive_params_growth_age Test 3: BMI SDS and percentile works (Z-score of -2, 0, 2, 5)", {
+  vs_data <- tibble::tribble(
+    ~USUBJID, ~SEX, ~AGECUR, ~AGEU, ~VSTESTCD, ~VSSTRESN,
+    "1001", "M", 2, "months", "BMI", 58,
+    "1002", "F", 578, "days", "BMI", 87,
+    "1003", "M", 2146, "days", "BMI", 13.5,
+    "1004", "F", 88.5, "months", "BMI", 21.9,
+  )
+
+  # pull WHO metadata for patients <2 years old, CDC metadata for patients >=2 years old
+  meta <- tibble::tribble(
+    ~SEX, ~AGE, ~AGEU, ~L, ~M, ~S,
+    "M", 61, "days", 0.1113, 16.3231, 0.08676,
+    "F", 578, "days", -0.5197, 15.6524, 0.08631,
+    "M", 2146, "days", -3.15039004, 15.37745304, 0.081706249,
+    "F", 2694, "days",-2.819657704, 15.56444426, 0.109308364,
+  )
+
+  actual <- derive_params_growth_age(
+    dataset = vs_data,
+    sex = SEX,
+    age = AGECUR,
+    age_unit = AGEU,
+    meta_criteria = meta,
+    parameter = VSTESTCD == "BMI",
+    analysis_var = VSSTRESN,
+    set_values_to_sds = exprs(
+      PARAMCD = "BMIASDS"
+    ),
+    set_values_to_pctl = exprs(
+      PARAMCD = "BMIAPCTL"
+    )
   )
 
+  vs_data_age <- vs_data %>%
+    mutate(AGECUR_c = case_when(
+      AGEU == "days"  ~ AGECUR,
+      AGEU == "months" ~ round(AGECUR*30.4375)),
+      AGEU_c = "days"
+    ) %>%
+    mutate(AGE = AGECUR_c, AGEU = AGEU_c) %>%
+    select(-ends_with("_c"),-AGECUR)
+
+  vs_data_meta <- vs_data_age %>%inner_join(meta, by=c("SEX", "AGE", "AGEU"))
+
+  vs_sds <- vs_data_meta %>%mutate(AVAL = (((VSSTRESN/M)^L)-1)/(L*S))
+  vs_pctl <- vs_sds %>% mutate(AVAL = pnorm(AVAL))
+  expected <- bind_rows(vs_sds, vs_pctl) %>% pull(AVAL)
+
+
   expect_equal(
     expected,
-    filter(actual, PARAMCD == "WT2AGEZ") %>% pull(AVAL)
+    filter(actual, PARAMCD %in% c("BMIASDS", "BMIAPCTL")) %>% pull(AVAL)
   )
 })
+
+
+
+## Test 4: Head circumference derivation works ----
+test_that("derive_params_growth_age Test 4: Head circumference SDS and percentile works", {
+  #LL: Up to 5 years old?
+  vs_data <- tibble::tribble(
+    ~USUBJID, ~SEX, ~AGECUR, ~AGEU, ~VSTESTCD, ~VSSTRESN,
+    "1001", "M", 2, "months", "HEADC", 39,
+    "1002", "F", 1157, "days", "HEADC", 50,
+  )
+
+  meta <- tibble::tribble(
+    ~SEX, ~AGE, ~AGEU, ~L, ~M, ~S,
+    "M", 61, "days", 1, 39.1349, 0.02997,
+    "F", 1157, "days",1, 48.6732, 0.02906,
+  )
+
+  actual <- derive_params_growth_age(
+    dataset = vs_data,
```

Comment:
Should the new argument `by_vars` be added there ?

Reply:
Thanks Fanny! Yes, `by_vars` added to all test cases.